src/main.c

```diff
@@ -31,25 +31,30 @@
 #include "sdi-session-helpers.h"
 #include "sdi-theme-monitor.h"
 
-static SnapdClient *client = NULL;
-static GtkApplication *app = NULL;
-static SdiThemeMonitor *theme_monitor = NULL;
-static SdiRefreshMonitor *refresh_monitor = NULL;
```

Comment:
By making these local they are a bit harder to manage, as they are never unreffed (same issue currently but all in one place). This would be better managed as a GApplication which I worked on at some point but never finished. Not sure if the in-between step of making them local is worth it?